roxie/ccd/ccdfile.cpp

```diff
@@ -361,6 +363,21 @@ class CLazyFileIO : public CInterface, implements ILazyFileIO, implements IDelay
         return current->size();
     }
 
+    virtual unsigned __int64 getStatistic(StatisticKind kind)
+    {
+        switch (kind)
+        {
+        case StTimeDiskReadIO:
+            return cycle_to_nanosec(getStatistic(StCycleDiskReadIOCycles));
+        case StTimeDiskWriteIO:
+            return cycle_to_nanosec(getStatistic(StCycleDiskWriteIOCycles));
+        }
+
+        CriticalBlock b(crit);
+        unsigned __int64 openValue = current ? current->getStatistic(kind) : 0;
+        return openValue + fileStats.getStatisticValue(kind);
```

Comment:
Not sure I quite follow the logic here, but should you be copying stats from current into fileStats if you background-close the file?